src/AutoRest.CSharp/Common/Generation/Types/TypeFactory.cs

```diff
@@ -40,55 +41,57 @@ public TypeFactory(OutputLibrary library, Type unknownType)
-                InputPrimitiveTypeKind.Uri or InputPrimitiveTypeKind.Url => new CSharpType(typeof(Uri), inputType.IsNullable),
-                InputPrimitiveTypeKind.Char => new CSharpType(typeof(char), inputType.IsNullable),
+                InputPrimitiveTypeKind.AzureLocation => new CSharpType(typeof(AzureLocation)),
+                InputPrimitiveTypeKind.Boolean => new CSharpType(typeof(bool)),
+                InputPrimitiveTypeKind.Bytes => Configuration.ShouldTreatBase64AsBinaryData ? new CSharpType(typeof(BinaryData)) : new CSharpType(typeof(byte[])),
+                InputPrimitiveTypeKind.ContentType => new CSharpType(typeof(ContentType)),
+                InputPrimitiveTypeKind.PlainDate => new CSharpType(typeof(DateTimeOffset)),
+                InputPrimitiveTypeKind.Decimal => new CSharpType(typeof(decimal)),
+                InputPrimitiveTypeKind.Decimal128 => new CSharpType(typeof(decimal)),
+                InputPrimitiveTypeKind.PlainTime => new CSharpType(typeof(TimeSpan)),
+                InputPrimitiveTypeKind.ETag => new CSharpType(typeof(ETag)),
+                InputPrimitiveTypeKind.Float32 => new CSharpType(typeof(float)),
+                InputPrimitiveTypeKind.Float64 => new CSharpType(typeof(double)),
+                InputPrimitiveTypeKind.Float128 => new CSharpType(typeof(decimal)),
+                InputPrimitiveTypeKind.Guid or InputPrimitiveTypeKind.Uuid => new CSharpType(typeof(Guid)),
+                InputPrimitiveTypeKind.Int8 => new CSharpType(typeof(sbyte)),
+                InputPrimitiveTypeKind.Int32 => new CSharpType(typeof(int)),
+                InputPrimitiveTypeKind.Int64 => new CSharpType(typeof(long)),
+                InputPrimitiveTypeKind.SafeInt => new CSharpType(typeof(long)),
+                InputPrimitiveTypeKind.Integer => new CSharpType(typeof(long)), // in typespec, integer is the base type of int related types, see type relation: https://typespec.io/docs/language-basics/type-relations
+                InputPrimitiveTypeKind.Float => new CSharpType(typeof(double)), // in typespec, float is the base type of float32 and float64, see type relation: https://typespec.io/docs/language-basics/type-relations
+                InputPrimitiveTypeKind.Numeric => new CSharpType(typeof(double)), // in typespec, numeric is the base type of number types, see type relation: https://typespec.io/docs/language-basics/type-relations
+                InputPrimitiveTypeKind.IPAddress => new CSharpType(typeof(IPAddress)),
+                InputPrimitiveTypeKind.ArmId => new CSharpType(typeof(ResourceIdentifier)),
+                InputPrimitiveTypeKind.ResourceType => new CSharpType(typeof(ResourceType)),
+                //InputPrimitiveTypeKind.RequestMethod => new CSharpType(typeof(RequestMethod)),
+                //InputPrimitiveTypeKind.ResourceIdentifier => new CSharpType(typeof(ResourceIdentifier)),
```

Comment:
we already have RequestMethod below and we should remove.
we should restore resource identifier here.